samples/highcharts/blog/nw-landing-page/demo.js

```diff
@@ -0,0 +1,1185 @@
+Math.easeInQuint =  function (pos) {
+    return Math.pow(pos, 5);
+};
+
+/*DONUT CHART*/
+const gases = ['Carbon Dioxide', 'Nitrogen', 'Oxygen', 'Argon', 'Methane',
+    'Sodium', 'Hydrogen', 'Helium', 'Other'];
+
+const planets = [
+    'Sun',
+    'Mercury',
+    'Venus',
+    'Earth',
+    'Moon',
+    'Mars',
+    'Jupiter',
+    'Saturn',
+    'Titan',
+    'Uranus',
+    'Neptune',
+    'Pluto'
+];
+
+const atmosphereData = [
+    [0, 0, 0, 0, 0, 0, 71, 26, 3],
+    [0, 0, 42, 0, 0, 22, 22, 6, 8],
+    [96, 4, 0, 0, 0, 0, 0, 0, 0],
+    [0, 78, 21, 1, 0, 0, 0, 0, 1],
+    [0, 0, 0, 70, 0, 1, 0, 29, 0],
+    [95, 2.7, 0, 1.6, 0, 0, 0, 0, 0.7],
+    [0, 0, 0, 0, 0, 0, 89.8, 10.2, 0],
+    [0, 0, 0, 0, 0, 0, 96.3, 3.2, 0.5],
+    [0, 97, 0, 0, 2, 0, 0, 0, 1],
+    [0, 0, 0, 0, 2.3, 0, 82.5, 15.2, 0],
+    [0, 0, 0, 0, 1.0, 0, 80, 19, 0],
+    [8, 90, 0, 0, 2, 0, 0, 0, 0]
+];
+
+let planetAtmosphere = [];
+
+function buildData(atmosphere) {
+    planetAtmosphere = [];
+    let count = 0;
+    const planetIndex = planets.findIndex(function (planet) {
+        return planet === atmosphere;
+    });
+    const tempAtmosphere = atmosphereData[planetIndex];
+    tempAtmosphere.forEach(function (item, index) {
+        planetAtmosphere.push([gases[index], item]);
+        count = count + 1;
+    });
+    return planetAtmosphere;
+}
+buildData('Mars');
+
+Highcharts.setOptions({
+    colors: [
+        Highcharts.getOptions().colors[4],
+        Highcharts.getOptions().colors[2],
+        Highcharts.getOptions().colors[0],
+        Highcharts.getOptions().colors[7],
+        Highcharts.getOptions().colors[3],
+        Highcharts.getOptions().colors[8],
+        Highcharts.getOptions().colors[6],
+        Highcharts.getOptions().colors[9],
+        Highcharts.getOptions().colors[1]
+    ]
+});
+
+Highcharts.chart('donut', {
+    chart: {
+        plotBackgroundColor: null,
+        plotBorderWidth: null,
+        plotShadow: false,
+        type: 'pie',
+        backgroundColor: 'transparent',
+        margin: 0,
+        spacing: 0
+    },
+    title: {
+        text: '',
+        y: 225
+    },
+    legend: {
+        enabled: false
+    },
+    exporting: {
+        enabled: false
+    },
+    credits: {
+        enabled: false
+    },
+    tooltip: {
+        enabled: false
+    },
+    plotOptions: {
+        pie: {
+            allowPointSelect: true,
+            cursor: 'pointer',
+            dataLabels: {
+                enabled: false
+            },
+            showInLegend: true
+        }
+    },
+    series: [{
+        name: 'Composition',
+        keys: ['name', 'y'],
+        colorByPoint: true,
+        innerSize: '70%',
+        data: planetAtmosphere
+    }]
+},
+function () {
+    const chart = this,
+        series = chart.series[0];
+    let newData = [];
+    let count = 0;
+    setInterval(function () {
```

Comment:
All the animations are quite processor heavy, so if an end user leaves this page open, the CPU is working hard. I propose one of two solutions:
a. Start the animations initially, and stop them after a fixed time when we assume the end user has scrolled down. Optionally we can start them again in a new sequence if the mouse/pointer interacts with the container div (just restart it with a mouseover event).
b. Use the [jQuery appear plugin](https://github.com/morr/jquery.appear), or something equivalent non-jQuery, to run the animations only when inside the viewport. But here too they should stop after a while in case the page is left open.

Reply:
I create a function that runs one time on load.